airbyte-integrations/connectors/destination-snowflake/src/main/kotlin/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeV1V2Migrator.kt

```diff
@@ -59,44 +71,44 @@ class SnowflakeV1V2Migrator(
         // The obvious database.getMetaData().getColumns() solution doesn't work, because JDBC
         // translates
         // VARIANT as VARCHAR
-        val columns =
-            database
-                .queryJsons(
+
+        try {
+            val showColumnsQuery =
                     """
-            SELECT column_name, data_type, is_nullable
-            FROM information_schema.columns
-            WHERE table_catalog = ?
-              AND table_schema = ?
-              AND table_name = ?
-            ORDER BY ordinal_position;
-            
-            """.trimIndent(),
-                    databaseName,
-                    namespace!!,
-                    tableName!!
-                )
+                       SHOW COLUMNS IN TABLE "$databaseName"."$namespace"."$tableName";
+                    """.trimIndent()
+            val showColumnsResult = database.queryJsons(
+                showColumnsQuery
+            )
+            val columnsFromShowQuery = showColumnsResult
                 .stream()
                 .collect(
                     { LinkedHashMap() },
                     { map: java.util.LinkedHashMap<String, ColumnDefinition>, row: JsonNode ->
-                        map[row["COLUMN_NAME"].asText()] =
+                        map[row["column_name"].asText()] =
                             ColumnDefinition(
-                                row["COLUMN_NAME"].asText(),
-                                row["DATA_TYPE"].asText(),
+                                row["column_name"].asText(),
+                                changeDataTypeFromShowQuery(ObjectMapper().readTree(row["data_type"].asText()).path("type").asText()),
```

Comment:
Why the double conversion with `.asText()` and then `readTree` ? `row["data_type"]` should already return a `JsonNode` so cast it to `ObjectNode` if it a struct.

Reply:
yeah, this seems odd